src/main/java/com/yelp/nrtsearch/server/luceneserver/BackupIndexRequestHandler.java

```diff
@@ -78,11 +96,122 @@ public BackupIndexResponse handle(IndexState indexState, BackupIndexRequest back
               indexName, backupIndexRequest.getServiceName(), backupIndexRequest.getResourceName()),
           e);
       return backupIndexResponseBuilder.build();
+    } finally {
+      if (snapshotId != null) {
+        releaseSnapshot(indexState, indexName, snapshotId);
+        deleteBackupIndicator();
+      }
```

Comment:
Can either of these operations fail and keep the lock from being released.

Reply:
I'm unlocking after the backup indicator file creation now and also before these lines in `finally`